moveit_ros/moveit_servo/demos/cpp_interface/demo_pose.cpp

```diff
@@ -0,0 +1,153 @@
+  auto servo_params = servo_param_listener->get_params();
+
+  // The publisher to send trajectory message to the robot controller.
+  rclcpp::Publisher<trajectory_msgs::msg::JointTrajectory>::SharedPtr trajectory_outgoing_cmd_pub =
+      demo_node->create_publisher<trajectory_msgs::msg::JointTrajectory>(servo_params.command_out_topic,
+                                                                         rclcpp::SystemDefaultsQoS());
+
+  // Create the servo object
+  auto planning_scene_monitor = createPlanningSceneMonitor(demo_node, servo_params);
+  auto servo = Servo(demo_node, servo_param_listener, planning_scene_monitor);
+
+  // Wait for some time, so that the planning scene is loaded in rviz.
+  // This is just for convenience, should not be used for sync in real application.
+  std::this_thread::sleep_for(std::chrono::seconds(3));
+
+  // For syncing pose tracking thread and main thread.
+  std::mutex pose_guard;
+  std::atomic<bool> stop_tracking = false;
+
+  // Set the command type for servo.
+  servo.expectedCommandType(CommandType::POSE);
+
+  // The dynamically updated target pose.
+  PoseCommand target_pose;
+  target_pose.frame_id = servo_params.ee_frame;
+  target_pose.pose = Eigen::Isometry3d::Identity();  // Target pose in ee frame (same as the ee frame pose)
+  // Convert the pose to planning frame.
+  tf2_ros::Buffer transform_buffer(demo_node->get_clock());
+  tf2_ros::TransformListener transform_listener(transform_buffer);
+  auto target_pose_msg =
+      convertIsometryToTransform(target_pose.pose, servo_params.planning_frame, target_pose.frame_id);
+  auto command_to_planning_frame = transform_buffer.lookupTransform(servo_params.planning_frame, target_pose.frame_id,
+                                                                    rclcpp::Time(0), rclcpp::Duration::from_seconds(2));
+  tf2::doTransform(target_pose_msg, target_pose_msg, command_to_planning_frame);
+  target_pose = PoseCommand{ servo_params.planning_frame, tf2::transformToEigen(target_pose_msg) };
+
+  // The pose tracking lambda that will be run in a separate thread.
+  auto pose_tracker = [&]() {
+    KinematicState joint_state;
+    rclcpp::WallRate tracking_rate(1 / servo_params.publish_period);
+    while (!stop_tracking && rclcpp::ok())
+    {
+      {
+        std::lock_guard<std::mutex> pguard(pose_guard);
+        joint_state = servo.getNextJointState(target_pose);
+      }
+      StatusCode status = servo.getStatus();
+      if (status != StatusCode::INVALID)
+        trajectory_outgoing_cmd_pub->publish(composeTrajectoryMessage(servo_params, joint_state));
+
+      tracking_rate.sleep();
+    }
+  };
+
+  // Pose tracking thread will exit upon reaching this pose.
+  Eigen::Isometry3d terminal_pose = target_pose.pose;
+  terminal_pose.translate(Eigen::Vector3d(0.0, 0.0, -0.1));
+
+  std::thread tracker_thread(pose_tracker);
+  tracker_thread.detach();
+
+  const double delta = 0.002;
```

Comment:
What is `delta`? Consider adding a comment and a more descriptive name explaining what this does -- which is dynamically updating the target pose so it's moving up in the +Z direction at a particular rate.

In fact, I might say bring up the entire `Eigen::Vector3d` defining the translation up in front of the loop so people can more easily change this value. Also, consider making it an `Isometry3d` instantiated using both translation and euler angles for the same reason. So people can change the demo to have the target pose rotate as well.

As an example, see:

```
Eigen::Isometry3d const goal_frame =
            Eigen::Translation3d(std::sin(M_PI_4), 3.0 * std::sin(M_PI_4), 0.0) *
            Eigen::AngleAxisd(0.75 * M_PI, Eigen::Vector3d::UnitZ());
```

Reply:
Done.